src/blocks/CalligraphyApp.tsx

```diff
@@ -0,0 +1,753 @@
+import Block from './Block'
+import { BlockModel } from './types'
+import './BlockStyles.css'
+import SeamSaveButton from '../components/SeamSaveButton';
+import { useEffect, useRef, useState } from 'react';
+
+import EditIcon from '@mui/icons-material/Edit';
+import BorderOuterIcon from '@mui/icons-material/BorderOuter';
+import UndoIcon from '@mui/icons-material/Undo';
+import CloseIcon from '@mui/icons-material/Close';
+import p5 from 'p5';
+import dots from "./assets/Calligraphy/dotsPreview.png"
+import grid from "./assets/Calligraphy/gridPreview.png"
+import lines from "./assets/Calligraphy/linesPreview.png"
+import brush1Brush from "./assets/Calligraphy/brushes/brush1.png";
+import sprayBrush from "./assets/Calligraphy/brushes/spray.png";
+import linesBrush from "./assets/Calligraphy/brushes/lines.png";
+import streakBrush from "./assets/Calligraphy/brushes/streak.png";
+import inkBrush from "./assets/Calligraphy/brushes/ink.png";
+import BlockFactory from './BlockFactory';
+
+const COLORS_DEFAULT = [
+  '#000000', '#333333',
+  '#666666', '#999999', '#cccccc', '#ffffff','#cdb4db', '#ffc8ddff', '#ffafccff', '#bde0feff', '#a2d2ffff', '#264653',
+  '#2a9d8f', '#e9c46a', '#f4a261', '#e76f51' ]
+
+// Component using a p5-wrapped canvas to draw
+interface CalligraphyCanvasProps {
+  width: string,
+  activeColor: string,
+  backgroundStyle: string
+  canvasClearSwitch: boolean
+  canvasUndoSwitch: boolean
+  currentBrush: string
+  setImageDataURL: (imgDataURL: string) => void
+}
+const CalligraphyCanvas = (props: CalligraphyCanvasProps) => {
+  const [p5Instance, setP5Instance] = useState<p5 | null>(null)
+  const [bufferInstance, setBufferInstance] = useState<p5.Graphics | null>(null)
+  const canvasDivRef = useRef<HTMLDivElement>(null)
+
+  /**A reference object accessible inside the p5 sketch */
+  const p5PassInRef = useRef<CalligraphyCanvasProps>({...props})
+  const canvasWidth = parseInt(props.width)
+  const ASPECT_RATIO = 1
+  
+  /** p5 Sketch Code That SHould Be Its Own File! */
```

Comment:
we're considering better ways to store seam miniapp projects; thanks for the feedback!